src/apps/clients/metadata.clj

```diff
@@ -54,14 +55,16 @@
                                       app-id))
 
 (defn filter-targets-by-ontology-search
-  [username category-attrs search-term app-ids]
-  (metadata-client/filter-targets-by-ontology-search (config/metadata-client)
-                                                     username
-                                                     (get-active-hierarchy-version)
-                                                     category-attrs
-                                                     search-term
-                                                     [app-target-type]
-                                                     app-ids))
+  [username category-attrs search-term app-ids & {:keys [validate :or {validate true}]}]
+  (if-let [active-hierarchy-version (get-active-hierarchy-version :validate validate)]
+    (metadata-client/filter-targets-by-ontology-search (config/metadata-client)
+                                                       username
+                                                       active-hierarchy-version
+                                                       category-attrs
+                                                       search-term
+                                                       [app-target-type]
+                                                       app-ids)
+    []))
```

Comment:
This change allows the app search service to work even if there's no active ontology set in the database, which prevents us from having to load an ontology specifically for apps integration tests.